go/arrow/compare_test.go

```diff
@@ -147,6 +150,37 @@ func TestTypeEqual(t *testing.T) {
 			},
 			false, true,
 		},
+		{
+			&StructType{
+				fields: []Field{
+					Field{Name: "f1", Type: PrimitiveTypes.Uint32, Nullable: true},
+				},
+				index: map[string]int{"f1": 0},
+			},
+			&StructType{
+				fields: []Field{
+					Field{Name: "f1", Type: PrimitiveTypes.Uint32, Nullable: true},
+					Field{Name: "f2", Type: PrimitiveTypes.Uint32, Nullable: true},
+				},
+				index: map[string]int{"f1": 0, "f2": 0},
```

Comment:
```suggestion
				index: map[string]int{"f1": 0, "f2": 1},
```
(I think. this should probably be fixed above as well.)